src/sentry/static/sentry/app/views/performance/trends/utils.tsx

```diff
@@ -326,34 +334,42 @@ export function transformEventStatsSmoothed(data?: Series[], seriesName?: string
       smoothedResults: undefined,
     };
   }
-  const currentData = data[0].data;
-  const resultData: SeriesDataUnit[] = [];
 
-  const smoothed = smoothTrend(currentData.map(({name, value}) => [Number(name), value]));
-
-  for (let i = 0; i < smoothed.length; i++) {
-    const point = smoothed[i] as any;
-    const value = point.y;
-    resultData.push({
-      name: point.x,
-      value,
-    });
-    if (!isNaN(value)) {
-      const rounded = Math.round(value);
-      minValue = Math.min(rounded, minValue);
-      maxValue = Math.max(rounded, maxValue);
+  const smoothedResults: Series[] = [];
+
+  for (const current of data) {
+    const currentData = current.data;
+    const resultData: SeriesDataUnit[] = [];
+
+    const smoothed = smoothTrend(
+      currentData.map(({name, value}) => [Number(name), value])
+    );
+
+    for (let i = 0; i < smoothed.length; i++) {
+      const point = smoothed[i] as any;
+      const value = point.y;
+      resultData.push({
+        name: point.x,
+        value,
+      });
+      if (!isNaN(value)) {
+        const rounded = Math.round(value);
+        minValue = Math.min(rounded, minValue);
+        maxValue = Math.max(rounded, maxValue);
+      }
     }
+    smoothedResults.push({
+      seriesName: seriesName || current.seriesName || 'Current',
```

Comment:
If a custom seriesName is provided, I think it should take precedence. It falls back to the seriesName for each series it's passed, it's likely with smoothing you'll want to pass a seriesName anyway. In this circumstance, using `current.seriesName` first wouldn't work for me because I want a specific smoothed name for each.